application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt

```diff
@@ -92,7 +92,10 @@ abstract class BackwardCompatibilityCheckBaseCommand : Callable<Unit> {
         ).filter {
             File(it).exists() && File(it).isValidSpec()
         }.toSet().also {
-            if(it.isEmpty()) exitWithMessage("No specs were changed, skipping the check.")
+            it.takeIf { it.isEmpty() }?.run {
```

Comment:
Just curious, why do we have to switch to `takeIf`? Can we not make the `if(it.isEmpty())` condition body a block body and just put the logger line and exitProcess with 0?

Reply:
Yes, just from a functional programming perspective I added but it's unclear and extra checks not needed so rectified it